db/db_impl_write.cc

```diff
@@ -1033,28 +1033,33 @@ Status DBImpl::SwitchWAL(WriteContext* write_context) {
   }
 
   auto oldest_alive_log = alive_log_files_.begin()->number;
-  auto oldest_log_with_uncommited_prep = FindMinLogContainingOutstandingPrep();
-
-  if (allow_2pc() &&
-      oldest_log_with_uncommited_prep > 0 &&
-      oldest_log_with_uncommited_prep <= oldest_alive_log) {
-    if (unable_to_flush_oldest_log_) {
+  bool flush_will_release_oldest_log = false;
+  if (allow_2pc()) {
+    auto oldest_log_with_uncommited_prep =
+        logs_with_prep_tracker_.FindMinLogContainingOutstandingPrep();
+
+    if (oldest_log_with_uncommited_prep > 0 &&
+        oldest_log_with_uncommited_prep <= oldest_alive_log) {
```

Comment:
if alive_log_files_ is the set of whole log files known to rocksdb, then how could oldest_log_with_uncommited_prep be possibility less than oldest_alive_log?

Reply:
It's existing code. I don't remember. It's probably won't Since it's not wrong to have "<=" I would keep it there. I dare not change it. Who knows whether changing it will trigger a bug.

Reply:
I rethought about it. We can take the risk. Let me change it to an assert.